hooks/useStore.js

```diff
@@ -142,8 +160,41 @@ const useStore = create(set => {
         Größe: 18cm`,
         RRPprice: 34.9,
         WSprice: 17.4,
+        quantity: 0,
+        sum: 0,
       },
     ],
+    addQuantity: id => {
+      set(state => {
+        return {
+          products: state.products.map(product =>
+            product.id === id
+              ? { ...product, quantity: Math.min(product.quantity + 1, 20) }
+              : product
+          ),
+        };
+      });
+    },
+    subtractQuantity: id => {
+      set(state => {
+        return {
+          products: state.products.map(product =>
+            product.id === id
+              ? { ...product, quantity: Math.max(product.quantity - 1, 0) }
+              : product
+          ),
+        };
+      });
+    },
+    removeQuantity: id => {
+      set(state => {
+        return {
+          products: state.products.map(product =>
+            product.id === id ? { ...product, quantity: 0 } : product
+          ),
+        };
+      });
+    },
```

Comment:
```suggestion
    setQuantity: (id, quantity) => {
      set(state => {
        return {
          products: state.products.map(product => 
            product.id === id ? { ...product, quantity } : product
          ),
        };
      });
    },
```

You can combine all three functions into one.